docs/build-and-publish-pipeline.md

```diff
@@ -0,0 +1,68 @@
+## Application Services Build and Publish Pipeline
+
+This document provides an overview of the build-and-publish pipeline used to make our work
+in this repo available to consuming applications. It's intended both to document the pipeline
+for development and maintenance purposes, and to serve as a basic analysis of the integrity
+protections that it offers (so you'll notice there are notes and open questions in place where
+we haven't fully hashed out all those details).
+
+The key points:
+
+* We use [Cargo](https://github.com/rust-lang/cargo) for building and testing the core Rust code in isolation,
+  [Gradle](https://gradle.org/) with [rust-android-gradle](https://github.com/mozilla/rust-android-gradle)
+  for combining Rust and Kotlin code into Android components and running tests against them,
+  and [Carthage](https://github.com/Carthage/Carthage) for combining Rust and Swift code into iOS components.
```

Comment:
Carthage doesn't do much really besides making distribution easier, the "combining" phase is all XCode.